src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1351,6 +1351,30 @@ public function findTaggedServiceIds(string $name, bool $throwOnAbstract = false
         return $tags;
     }
 
+    /**
+     * Find definitions of value-object classes by tag, and add the "container.excluded" tag to them.
+     *
+     * @param bool $autoExclude Adds the "container.excluded" tag to the returned definitions
+     *
+     * @return array<string, array> An array of tags with the tagged classes as key, holding a list of attribute arrays
+     */
+    public function findTaggedValueClasses(string $name, bool $autoExclude = true): array
+    {
+        $this->usedTags[] = $name;
+        $tags = [];
+        foreach ($this->getDefinitions() as $id => $definition) {
+            if ($definition->hasTag($name)) {
+                if ($autoExclude) {
+                    $definition->addTag('container.excluded', ['source' => \sprintf('by tag "%s"', $name)]);
```

Comment:
let's also make the definition abstract as we do in FileLoader?
```suggestion
                    $definition->setAbstract(true)
                        ->addTag('container.excluded', ['source' => \sprintf('by tag "%s"', $name)]);
```